ocaml/libs/uuid/uuidx.mli

```diff
@@ -33,7 +33,14 @@ val null : 'a t
 val make : unit -> 'a t
 (** Create a fresh UUID *)
 
+val make_uuid_fast : unit -> 'a t
```

Comment:
Hmm I'd need a way to negate `'session', in the polymorphic variant, but I don't think OCaml's type system supports constraints like that. I can define upper or lower bounds that would include a conflicting `'session of 'somethingbad`, but that fails when an exact Ref.t is required.

It is quite easy to declare that make_secret returns a 'session Ref.t, but I'll keep thinking on whether there is a way to say that 'make' returns an 'a that can be anything exception 'session.

Reply:
I found a way, and it has also found a bug: the local session must also use the urandom method for generating the secret.

(had to tweak a few other things in mlis and testcases too).

There aren't that many "classes" in XAPI, so I've enumerated them all (by hand for now) in Uuidx.mli, and ensured that session cannot be present there (by adding session with an incompatible type, so that when unified it prints a nice error message):
```
    ```
    File "ocaml/tests/test_client.ml", line 18, characters 47-57:
    18 |       Test_common.make_session ~__context ~ref:session_id
                                                        ^^^^^^^^^^
    Error: This expression has type [< Uuidx.not_session ] Ref.t
           but an expression was expected of type [ `session ] Ref.t
           Type [< Uuidx.not_session ] = [< `session of [ `use_make_uuid_urnd ] ]
           is not compatible with type [ `session ]
           Types for tag `session are incompatible
    ```
```

Reply:
@mg12 you were also asking whether we can use the typesystem for tracking secrets, see above.

Reply:
Does this mean that adding a new class to XAPI would require adding it to these two places too? Do we not produce some list of classes automatically? There's already a bit of boilerplate that needs to be written with the new class, and I don't think an error will show up here if one forgets to add the class.

Reply:
I've changed a bit how the parameter is handled, now there is a type constraint on `'a`, and it must be one of the known classes. If we add a new class then the generated db_actions.ml will ensure we get a type error, because it'll try to use a type parameter that is not in one of the known ones.

Eventually it'd be nice if we could autogenerate this list of classes, although we add classes rare enough.